ant-networking/src/cmd.rs

```diff
@@ -1026,46 +1037,64 @@ impl SwarmDriver {
 
         if *is_bad {
             warn!("Cleaning out bad_peer {peer_id:?}. Will be added to the blocklist after informing that peer.");
-            if let Some(dead_peer) = self.swarm.behaviour_mut().kademlia.remove_peer(&peer_id) {
-                self.update_on_peer_removal(*dead_peer.node.key.preimage());
-            }
+            if let Some(bad_behaviour) = new_bad_behaviour {
+                let is_connection_issue =
+                    matches!(bad_behaviour, NodeIssue::CriticalConnectionIssue);
+
+                // If it is a connection issue, then we should not block it completely, give it more chances.
+                // Since we still keep the Vec<issue> around, a new connection issue will again result in RT eviction,
+                // the node will be evicted time after time.
+                // So we'll set is_bad to false to give chance to faulty connected nodes.
+                if is_connection_issue {
+                    warn!("Peer {peer_id:?} is considered as bad due to connection issue. Will be given more chances.");
+                    *is_bad = false;
```

Comment:
it's already inside the `if *is_bad {` clause ?
change `is_bad` doesn't make any sense.
Also, the following `record_metrics` and `ConsidedAsBad` block will be executed as well?
I think here we shall just `remove peer` then return ?